### PR TITLE
TDI-43660 : pubsub connection - pattern for project name

### PR DESCRIPTION
**Pull request by rlecomte-talend**
White spaces forbidden in project name

### CODE REVIEW

File: pubsub/src/main/java/org/talend/components/pubsub/datastore/PubSubDataStore.java [hunk 1]
```diff
@@ -39,6 +40,7 @@
     @Option
     @Required
     @Documentation("Google Cloud Platform Project")
+    @Pattern("/^\\S+$/")
```
Comment (undx) [inline pubsub/src/main/java/org/talend/components/pubsub/datastore/PubSubDataStore.java]: Everything except space chars, very flexible. No constraints on project's names? (for instance, I know that azure has some...)

Comment (rlecomte-talend) [reply pubsub/src/main/java/org/talend/components/pubsub/datastore/PubSubDataStore.java]: In fact, I don't even know if whitespaces are forbidden. Documentation is not very explicit : https://cloud.google.com/resource-manager/docs/creating-managing-projects#before_you_begin

Comment (undx) [reply pubsub/src/main/java/org/talend/components/pubsub/datastore/PubSubDataStore.java]: So, maybe just keep the required annotation.

Comment (rlecomte-talend) [reply pubsub/src/main/java/org/talend/components/pubsub/datastore/PubSubDataStore.java]: Another source : https://edu.google.com/openonline/course-builder/docs/1.11/set-up-course-builder/create-a-cloud-project.html
Explicitly says no whitespace, and thta's the only rule given.

Comment (undx) [reply pubsub/src/main/java/org/talend/components/pubsub/datastore/PubSubDataStore.java]: OK, I read : 
> Project Name
> To be valid, a project name must
> be between 4 and 30 characters
> consist of letters, numbers, quotes, hyphens, spaces or exclamation point

Then 
> Project ID
> The project ID is a single word (no spaces) that uniquely identifies your application

So, we're talking about the project ID and not the project name. 
The label / fieldname is confusing therefore.
Maybe you should rename the option.

Also, are those kind of chars supported `#$%^&%*()`...  ?

Comment (rlecomte-talend) [reply pubsub/src/main/java/org/talend/components/pubsub/datastore/PubSubDataStore.java]: Already talked about it durgin development. The Google client API uses "project name" to talk about this ID... It's quite of a mess. We chose to keep it like that for now.

I also read
> The default project ID is composed of the project name and a randomly assigned number. You can customize the project ID only during project creation.
So, if project name can contains space, so can project id...

Comment (rlecomte-talend) [reply pubsub/src/main/java/org/talend/components/pubsub/datastore/PubSubDataStore.java]: @fdesiles made some more tests by creating new project. The rules are not the same on the project creation page... He's adding a screenshot

Comment (rlecomte-talend) [reply pubsub/src/main/java/org/talend/components/pubsub/datastore/PubSubDataStore.java]: "Project ID can have lowercase letters, digits or hyphens and must start with a lowercase letter". 

I'll change the regex to match this.

**Pull request merged**